File: src/zork/models/entities/Stat.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package zork.models.entities;

/**
 *
 * @author d.peters
 */
public class Stat {
    
    /**
     * 
     */
    private double value;
    
    /**
     * 
     */
    private double minValue;
    
    /**
     * 
     */
    private double maxValue;
    
    public Stat(double value, double minValue, double maxValue){
        this.value = value;
        this.minValue = minValue;
        this.maxValue = maxValue;
    }

    /**
     * 
     * @return 
     */
    public double getValue() {
        return value;
    }
    
    /**
     * 
     * @param add 
     */
    public void increase(double add){
        if (this.value + add > this.maxValue){
            this.value += add;
        }
    }
    
    /**
     * 
     * @param subtract 
     */
    public void decrease(double subtract){
        this.value -= subtract;
    }
    
}

File: src/main/kotlin/ch/peters/daniel/zork/models/entities/base/Stat.kt
package ch.peters.daniel.zork.models.entities.base

/**
 * Container object for player stat.
 *
 * @author Daniel Peters <daniel.peters.ch@gmail.com>
 *
 * @version 1.1.0
 */
class Stat(var value: Double, var minValue: Double, var maxValue: Double) {
  /**
   * Increase the current value of this object.
   *
   * @param add added amount
   */
  fun increase(add: Double) {
    if (value + add > maxValue) {
      value += add
    }
  }

  /**
   * Decrease the current value of this object.
   *
   * @param subtract reduced amount
   */
  fun decrease(subtract: Double) {
    // Check because armor could make damage negative (which would heal char)
    if (subtract > 0) {
      value -= subtract
    }
  }
}
